templates/activity-stream/on-save-recs/on-save-recs.1.schema.json

```diff
@@ -0,0 +1,51 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "title": "on-save-recs",
+  "properties": {
+    @ACTIVITY-STREAM_IMPRESSIONID_1_JSON@,
+    @ACTIVITY-STREAM_EXPERIMENTS_1_JSON@,
+    "action_position": {
+      "description": "A zero based integer indicating the position of this event",
+      "type": "integer"
+    },
+    "addon_version": {
+      "type": "string"
+    },
+    "event": {
+      "enum": [
+        "impression",
+        "click",
+        "save"
+      ],
+      "description": "An event identifier",
+      "type": "string"
+    },
```

Comment:
Have you considered replacing this with an `events` array with `action` and `position` fields to allow multiple impressions to be expressed in a single ping? I'm assuming that we generally present a series of impressions all at once, so we could make a single call. That would save some bandwidth for the client, reduce the raw storage size, and cut down on some per-ping overhead in the pipeline. We could provide a view that performs the `UNNEST` necessary to transform it into this flattened representation.

This has been discussed a bit for previous A-S pings and I believe was rejected because it would have required additional code changes, and in those cases we were replacing existing ping-centre calls with identical payloads to the telemetry pipeline. If this is going to be new code on the client side, though, allowing multiple events per ping might make sense.

Reply:
Sounds reasonable, making this change now.